File: cpp/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.length()==1)
            return 1;
        
        int j,max=0,count=0,found;   
        string ss="";
        for(j=0;j<s.length();j++){
            found=ss.find(s[j]);
            if(found<0){
                count++;
                ss+=s[j];     
            }else{
                ss.erase(0,found+1);
                ss+=s[j]; 
                count=ss.length();
            }
               max=std::max(max,count);
        }
        return max;
    }
};

File: 3-Longest-Substring-Without-Repeating-Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        charSet = {}
        l,r = 0
        res = 0
        
        while r < len(s):
            if s[r] in d:
                l = max(l, d[s[r]] + 1)
            d[s[r]] = r
            res = max(res, r - l + 1)
            r += 1
        return res